### PR TITLE
Add page transition progress indicator

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Show the progress indicator on the top of the page during page transitions.

## Justify why they are needed

To make the page transitions more smooth.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Review by alebedev — APPROVED** [review]
Great visual improvement!

File: apps/store/src/services/nprogress/nprogress.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    router.events.on('routeChangeError', handleRouteChangeComplete)
+
+    return () => {
+      router.events.off('routeChangeStart', handleRouteStart)
+      router.events.off('routeChangeComplete', handleRouteChangeComplete)
+      router.events.off('routeChangeError', handleRouteChangeComplete)
+    }
+  }, [router.events])
+
+  return <Global styles={[styles, { ':root': { '--nprogress-color': color } }]} />
```
Comment (gustaveen) [inline apps/store/src/services/nprogress/nprogress.tsx:34]: Oh smart to set it as a css variable to avoid hydration errors! 💡 Will use it for the links 🙌

**Review by gustaveen — APPROVED** [review]
Stylish! 💅

File: apps/store/src/services/nprogress/nprogress.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/services/nprogress/nprogress.tsx:34]: Yeah, it's too bad that emotion doesn't make them easier to use but this does seem like a good implementation :)

**Pull request merged**